sdk/scheduler/src/main/java/com/mesosphere/sdk/dcos/CertificateAuthorityClient.java

```diff
@@ -0,0 +1,36 @@
+package com.mesosphere.sdk.dcos;
+
+import com.mesosphere.sdk.dcos.ca.CAException;
+
+import java.io.IOException;
+import java.security.cert.CertificateException;
+import java.security.cert.X509Certificate;
+import java.util.Collection;
+
+/**
+ * Represents abstraction over DC/OS Certificate Authority.
+ * @see "https://docs.mesosphere.com/1.9/networking/tls-ssl/ca-api/"
+ */
+public interface CertificateAuthorityClient {
+
+    /**
+     * Create a new certificate from CSR by contacting certificate authority.
+     *
+     * @param csr
+     * @return
+     * @throws IOException
+     * @throws CertificateException
+     */
+    X509Certificate sign(byte[] csr) throws IOException, CertificateException, CAException;
```

Comment:
Are any callers expected to behave differently depending on which of these exceptions is thrown? E.g. looks like CAException is only being checked in tests. IMO this could just be `throws Exception` or `throws IOException` (with CertificateException/CAException extending IOException in the latter case).